rskj-core/src/test/java/co/rsk/net/discovery/DiscoveredPeersPersistenceServiceTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2024 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.net.discovery;
+
+import org.ethereum.util.SimpleFileWriter;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+class DiscoveredPeersPersistenceServiceTest {
+
+    private DiscoveredPeersPersistenceService service;
+    private PeerExplorer peerExplorer;
+    private Path filePath;
+    private SimpleFileWriter fileDataSaver;
+
+    @BeforeEach
+    void setup() throws IOException {
+        peerExplorer = mock(PeerExplorer.class);
+        filePath = Files.createTempFile("test", ".txt");
+        fileDataSaver = mock(SimpleFileWriter.class);
+        service = new DiscoveredPeersPersistenceService(peerExplorer, filePath,fileDataSaver);
+    }
+
+    @Test
+    void testStop() throws IOException {
+        List<String> knownPeers = Arrays.asList("peer1", "peer2", "peer3");
+        when(peerExplorer.getKnownHosts()).thenReturn(knownPeers);
+
+        service.stop();
+
+        verify(fileDataSaver, times(1)).saveDataIntoFile(anyString(), any(Path.class));
```

Comment:
Wdyt if we verify the fields exactly instead of any input?
You have the filePath that it would try to save and you have the input peers mocked, you could check it instead of any:

`        verify(fileDataSaver, times(1)).saveDataIntoFile(eq("peer1\npeer2\npeer3\n"), eq(filePath));`

would do the job.

Reply:
Yep it is a good idea!